cmd/soroban-rpc/internal/ledgerentry_storage/db.go

```diff
@@ -180,6 +187,70 @@ func (s *sqlDB) GetLedgerEntry(key xdr.LedgerKey) (xdr.LedgerEntry, bool, uint32
 	return entry, true, seq, nil
 }
 
+// TODO : move these Scan() and Value() functions to xdr/db.go in the monorepo
+type ledgerCloseMeta xdr.LedgerCloseMeta
+
+// Scan reads from src into a ledgerCloseMeta  struct
+func (t *ledgerCloseMeta) Scan(src any) error {
+	return safeBase64Scan(src, t)
+}
+
+// Value implements the database/sql/driver Valuer interface.
+func (c ledgerCloseMeta) Value() (driver.Value, error) {
+	return xdr.MarshalBase64(c)
+}
+
+// safeBase64Scan scans from src (which should be either a []byte or string)
+// into dest by using `SafeUnmarshalBase64`.
+func safeBase64Scan(src, dest any) error {
+	var val string
+	switch src := src.(type) {
+	case []byte:
+		val = string(src)
+	case string:
+		val = src
+	default:
+		return fmt.Errorf("Invalid value for %T", dest)
+	}
+
+	return xdr.SafeUnmarshalBase64(val, dest)
+}
+
+// GetAllLedgers returns all ledgers in the database.
+func (s *sqlDB) GetAllLedgers() ([]xdr.LedgerCloseMeta, error) {
+	sqlStr, args, err := sq.Select("meta").From(ledgerCloseMetaTableName).OrderBy("sequence asc").ToSql()
+	if err != nil {
+		return nil, err
+	}
+	var results []ledgerCloseMeta
+	err = s.db.Select(&results, sqlStr, args...)
+	ledgers := make([]xdr.LedgerCloseMeta, len(results))
+	for i, ledger := range results {
+		ledgers[i] = xdr.LedgerCloseMeta(ledger)
+	}
+	return ledgers, err
+}
+
+// GetLedger fetches a single ledger from the db.
+func (s *sqlDB) GetLedger(sequence uint32) (xdr.LedgerCloseMeta, bool, error) {
```

Comment:
before sql gets more verbose, is it worth considering a type-safe compile time binding like go-jet.